### PR TITLE
[+] added possibility to create up to 5 pool resources

### PR DESCRIPTION
**Pull request by dthmrphn**

### CODE REVIEW

File: src/mem/mem_it.c [hunk 1]
```diff
@@ -3,20 +3,38 @@
...
 static const mem_desc_t *m_desc_;
 
-int mem_init(const void *desc, const void *params) {
+/**
+ * @brief creates memory resource
+ * 
+ * @param desc pointer to allocator descriptor
+ * @param params pointer to allocator params
+ * @return 0 if success 
+ */
```
[the remark this answers is not in the document]
Comment (uladzislaudd) [reply src/mem/mem_it.c:12]: not trying to be annoying, but can you make return descriptions like [here](https://man7.org/linux/man-pages/man3/getcwd.3.html).

**Review by uladzislaudd — APPROVED** [review]
LGFM!

**Comment by uladzislaudd** [comment]
@dthmrphn How do you run your tests? Are you using Valgrind?

File: include/mem.h [hunk 2]
```diff
@@ -13,10 +13,11 @@ typedef struct poolparam {
     size_t count;   /* count of chunks */
 } p_pool_t;
 
-int   mem_init(const void *desc, const void *params);
+int   mem_create(const void *desc, const void *params);
```
[the remark this answers is not in the document]
Comment (dthmrphn) [reply include/mem.h:16]: this function also creates another memory resource so init doesn't sound correct

File: src/mem/mem_it.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (dthmrphn) [reply src/mem/mem_it.c:12]: might make it later with error code handling

File: src/mem/mem_pool_p.c [hunk 3]
```diff
@@ -4,29 +4,51 @@
...
     uint8_t data[]; /* pointer to allocated block   */
 } mblock_t;
 
 /* holding allocator's params there */
-static p_pool_t params_;
+static p_pool_t params_[MEM_POOL_RESOURCE_COUNT];
+static size_t last_added = 0;
 
 /* should return error ?*/
 static int m_pool_init(const void *params) {
```
[the remark this answers is not in the document]
Comment (dthmrphn) [reply src/mem/mem_pool_p.c:21]: this static function have short names, they should not be in external interface

File: include/mem.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (uladzislaudd) [reply include/mem.h:16]: sounds reasonable

File: src/mem/mem_it.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (uladzislaudd) [reply src/mem/mem_it.c:12]: sure, keep it cool

File: src/mem/mem_pool_p.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (uladzislaudd) [reply src/mem/mem_pool_p.c:21]: sounds reasonable

**Pull request merged**